.github/workflows/release.yml

```diff
@@ -746,9 +746,33 @@ jobs:
       with:
         arch: ${{ matrix.host }}
 
+    - name: Save code signing certificate to disk
+      id: save_cert
+      uses: timheuer/base64-to-file@v1
+      with:
+        filename: 'openttd.p12'
+        # The certificates in a PKCS12 file encoded as a base64 string
+        encodedString: ${{ secrets.WIN_CERTIFICATE_P12 }}
+      # If this is run on a fork, there may not be a certificate set up - continue in this case
+      continue-on-error: true
+
+    - name: Import code signing certificate
+      shell: powershell
+      # If this is run on a fork, there may not be a certificate set up - continue in this case
+      continue-on-error: true
+      env:
+        WIN_CODESIGN_P12: ${{ steps.save_cert.outputs.filePath }}
+        WIN_CODESIGN_PWD: ${{ secrets.WIN_CERTIFICATE_PASSWORD }}
+
+      run: |
+        $pwd = ConvertTo-SecureString $env:WIN_CODESIGN_PWD -AsPlainText -Force
+        Import-PfxCertificate -FilePath $env:WIN_CODESIGN_P12 -CertStoreLocation Cert:\CurrentUser\My -Password $pwd
+
     - name: Build (with installer)
       if: needs.source.outputs.is_tag == 'true'
       shell: bash
+      env:
+        WIN_CODESIGN_CN: ${{ secrets.WIN_CERTIFICATE_COMMON_NAME }}
```

Comment:
Please move `env` block below `run` block (all other places do it like that, with the exception of your MacOS work :P I spotted that way too late, sorry :D)